shadowsocks/tcp.go

```diff
@@ -62,51 +44,56 @@ func remoteIP(conn net.Conn) net.IP {
 	return nil
 }
 
-func findAccessKey(clientConn onet.DuplexConn, cipherList CipherList) (string, onet.DuplexConn, error) {
-	// This must have enough space to hold the salt + 2 bytes chunk length + AEAD tag (Oeverhead) for any cipher
-	replayBytes := make([]byte, 0, 32+2+16)
+func (s *tcpService) findAccessKey(clientConn onet.DuplexConn) (string, onet.DuplexConn, error) {
```

Comment:
I'd rather keep this function outside tcpService, so it's clear it doesn't need the internal state, and move the IV test to outside, decoupling the concerns.

My suggestion: revert `findAccessKey`, and change it to return the IV. Then you check the IV in the caller. This will keep the concerns separate.

Reply:
Done